### ISSUE TITLE
LightDM doesn't show any user on the list

### ISSUE THREAD

**Issue by nawordar** [utdemir/dotfiles#52]
Hi, I forked your configuration and removed the `services.xserver.displayManager.lightdm.autoLogin` part from `system.nix` to disable auto login.
Now the user list in LightDM is empty, and I need to enter username manually. When I installed and configured nixOS from scratch, my username was on the list. Do you know what may be the reason?

**Comment by 3rly** [comment]
That is because it pulls the user list off the user.nix file. One thing I would try if I was you, is to add extra users to that file (user.nix) and see if that resolves it. OR add a second user file (i.e. user2.nix), fill the same info and then reference that file in your imports. I would do the latter, since it allows for better modularity. For example:

```sh
imports =
    (builtins.filter builtins.pathExists [ ./system-private.nix ]) ++ [
      ./user.nix] ++ [user2.nix];
```

Then modify the `services.xserver` line:
```sh  
services.xserver = {
    enable = true;
    autorun = true;
    desktopManager.xterm.enable = true;
    displayManager.lightdm = {
      enable = true;
      greeter.enable = false;
      autoLogin = {
        enable = false;
        user = config.dotfiles.username;
        user2 = config.dotfiles.username; # new user
      };
    };
    xkbOptions = "ctrl:swapcaps";
  };
```
I am not 100% certain this would work, since I haven't tested this recommendation. I have a different use case. Good luck.

**Comment by 3rly** [comment]
I forgot to add that this line also will need to be updated:

`trustedUsers = [ "root" config.dotfiles.username ];`

**Comment by utdemir** [comment]
Hey @nawordar , sorry I didn't notice the issue earlier. I reproduced your issue by disabling autologin on my system. Unfortunately I can not think of a reason why the user list is empty. I usually use auto-login since I already use a full disk encryption and am the only user of the computer.

I checked the source code of lightdm module, and there seems to be a few different ways to hide an user[1], however none of them applies to us (our uid is greater than 500, and we're not hiding ourselves explicitly). I tried anoter greeter, and created a few test users with a few different settings, and none of them appeared on the greeter.

Unfortunately this requires more investigation and you'd understand that I don't have many incentives to pursue it much longer. But I'm happy to answer any other questions.

@3rly Actually we are not the ones populating `lightdm`'s user list; we just create an ordinary user and expect lightdm to find it. So I think there is something which makes `lightdm` not show our current user, but I can not figure out why.

[1]: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/x11/display-managers/lightdm.nix#L34-L37
